test/unix_socket/main.c

```diff
@@ -65,6 +65,67 @@ int create_connceted_sockets_default(int *sockets) {
     return create_connected_sockets(sockets, "unix_socket_default_path");
 }
 
+int create_connected_sockets_then_rename(int *sockets) {
+    char *socket_original_path = "/tmp/socket_tmp";
+    char *socket_ready_path = "/tmp/.socket_tmp";
+    int listen_fd = socket(AF_UNIX, SOCK_STREAM, 0);
+    if (listen_fd == -1) {
+        THROW_ERROR("failed to create a unix socket");
+    }
+
+    struct sockaddr_un addr;
+    memset(&addr, 0, sizeof(struct sockaddr_un)); //Clear structure
+    addr.sun_family = AF_UNIX;
+    strcpy(addr.sun_path, socket_original_path);
+    socklen_t addr_len = strlen(addr.sun_path) + sizeof(addr.sun_family);
+    if (bind(listen_fd, (struct sockaddr *)&addr, addr_len) == -1) {
+        close(listen_fd);
+        THROW_ERROR("failed to bind");
+    }
+
+    if (listen(listen_fd, 5) == -1) {
+        close(listen_fd);
+        THROW_ERROR("failed to listen");
+    }
+
+    // rename to new path
+    unlink(socket_ready_path);
+    if (rename(socket_original_path, socket_ready_path) < 0) {
+        THROW_ERROR("failed to rename");
+    }
+
+    int client_fd = socket(AF_UNIX, SOCK_STREAM, PF_UNIX);
```

Comment:
The third argument `protocol` of the `socket` API should be given a value of 0, indicating that you want the default protocol. There is no other protocols available in the unix socket family.




Here is the relevant paragraphs in man pages:

> Normally only a single protocol exists to support a
       particular socket type within a given protocol family, in which
       case protocol can be specified as 0.

> #include <sys/socket.h>
> #include <sys/un.h>
> unix_socket = socket(AF_UNIX, type, 0);

> The manifest constants used under 4.x BSD for protocol families
> are PF_UNIX, PF_INET, and so on, while AF_UNIX, AF_INET, and so
> on are used for address families.  However, already the BSD man
>  page promises: "The protocol family generally is the same as the
>  address family", and subsequent standards use AF_* everywhere.

And `PF_UNIX` is just an alias of `AF_UNIX` or `AF_LOCAL`. So both `PF_UNIX` and `AF_UNIX` are meant to be used for the first argument of `family`. 

```c
// File include/sys/socket.h in musl libc

#define PF_LOCAL        1
#define PF_UNIX         PF_LOCAL
#define PF_FILE         PF_LOCAL
//...

#define AF_LOCAL        PF_LOCAL
#define AF_UNIX         AF_LOCAL
#define AF_FILE         AF_LOCAL
//...
```

Reply:
Done. Thanks for your review.